File: Data Structures & Algorithms/daily-temperatures/submission-2.cpp
class Solution
{
public:
    vector<int> dailyTemperatures(vector<int> &temperatures)
    {
        stack<tuple<int, int> > s;
        vector<int> res(temperatures.size());
        for (int i = 0; i < temperatures.size(); i++)
        {
            if (s.size() > 0)
            {
                int stackCount = s.size();
                for (int j = 0; j < stackCount; j++)
                {
                    auto [temp, index] = s.top();

                    if (temperatures[i] > temp)
                    {
                        s.pop();
                        res[index] = i - index;
                    }
                }
            }
            s.push(make_tuple(temperatures[i], i));
        }

        return res;
    }
};

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.cs
public class Solution
{
    public int EvalRPN(string[] tokens)
    {
        Stack<int> stack = new Stack<int>();

        foreach (var s in tokens)
        {
            if (int.TryParse(s, out var i))
            {
                stack.Push(i);
            }
            else
            {
                int temp = stack.Pop();
                switch (s)
                {
                    case "+":
                        temp += stack.Pop();
                        stack.Push(temp);
                        break;
                    case "-":
                        temp = stack.Pop() - temp;
                        stack.Push(temp);
                        break;
                    case "*":
                        temp = stack.Pop() * temp;
                        stack.Push(temp);
                        break;
                    case "/":
                        temp = stack.Pop() / temp;
                        stack.Push(temp);
                        break;
                }
                Console.WriteLine(temp);
            }
        }

        return stack.Pop();
    }
}